packages/legend-application-studio/src/stores/editor/sidebar-state/BulkServiceRegistrationState.ts

```diff
@@ -48,54 +82,159 @@ export class BulkServiceRegistrationState extends ServiceConfigState {
     );
 
     makeObservable(this, {
-      showSuccessModel: observable,
+      editorStore: false,
+      service: observable,
+      registrationResult: observable,
+      setShowFailingView: action,
+      setSelectService: action,
+      isSelected: observable,
+    });
+    this.globalBulkServiceRegistrationState =
+      globalBulkServiceRegistrationState;
+    this.service = service;
+  }
+
+  setSelectService(): void {
+    this.isSelected ? (this.isSelected = false) : (this.isSelected = true);
+  }
+  setShowFailingView(val: boolean): void {
+    this.showFailingView = val;
+  }
+
+  handleRegistrationResult(
+    registrationResult: ServiceRegistrationResult,
+  ): void {
+    try {
+      assertTrue(registrationResult.service?.pattern === this.service.pattern);
```

Comment:
why do we need this assert ?

Reply:
I just added this as a sanity check to make sure the service that we get from bulkRegistrationState patches the current instance of the service

I also saw that it was being done in GlobalTestRunner in handleTestableResult